imgee/storage.py

```diff
@@ -216,7 +222,7 @@ def resize_and_save(img, size, is_thumbnail=False):
     Get the original image from local disk cache, download it from S3 if it misses.
     Resize the image and save resized image on S3 and size details in db.
     """
-    src_path = download_frm_s3(img.name + img.extn)
+    src_path = download_from_s3(img.filename)
```

Comment:
Oh, I get it. We're not storing the exact name in s3 in db. `orig_extn` is the original extension of the file. but as the thumbnail extension can be different from original extension, that target extension is stored in the `ALLOWED_MIMETYPES` dict, and taken from there every time we request for the extension.